low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -269,6 +270,66 @@ def prepare_mem_json(self, status, description):
         }
         return memory_dict
 
+    def get_nw_ports_info(self):
+        """Return the Network ports information."""
+        network_cable_data = []
+        io_counters = psutil.net_io_counters(pernic=True)
+
+        for interface, addrs in psutil.net_if_addrs().items():
+            nic_info = self.get_health_template(interface, False)
+
+            specifics = {}
+            for addr in addrs:
+                if addr.family == socket.AF_INET:
+                    specifics["ipV4"] = addr.address
+
+            if interface in io_counters:
+                io_info = io_counters[interface]
+                specifics.update({
+                    "networkErrors": io_info.errin + io_info.errout,
+                    "droppedPacketsIn": io_info.dropin,
+                    "droppedPacketsOut": io_info.dropout,
+                    "packetsIn": io_info.packets_recv,
+                    "packetsOut": io_info.packets_sent,
+                    "trafficIn": io_info.bytes_recv,
+                    "trafficOut": io_info.bytes_sent
+                })
+
+            nw_status, nw_cable_conn_status = \
+                self.get_nw_status(interface)
+
+            specifics["nwStatus"] = nw_status
+            specifics["nwCableConnStatus"] = nw_cable_conn_status
+
+            # set health status, description and recommendation
+            map_status = {"UP": "OK", "DOWN": "Disabled/Failed",
+                          "UNKNOWN": "NA"}
+            health_status = map_status[nw_cable_conn_status]
+
+            desc = "Network Interface '%s' is %sin good state" % (
+                interface, '' if health_status == "OK" else 'not '
+            )
+
+            self.set_health_data(nic_info, health_status, description=desc,
+                                 specifics=[specifics])
+
+            network_cable_data.append(nic_info)
+
+        return network_cable_data
+
+    def get_nw_status(self, interface):
+        """Read & Return the latest network status from sysfs files."""
+        try:
+            nw_status = self.sysfs.fetch_nw_operstate(interface)
+            nw_cable_conn_status = self.sysfs.fetch_nw_cable_status(
+                self.sysfs.get_sys_dir_path('net'), interface
+            )
+        except Exception as err:
+            raise ResourceMapError(errno.EINVAL,
```

Comment:
This will affect returning other FRU health infos even when this one fails. Please handle this in `get_health_info`

Reply:
as discussed with @cdeshmukh , I am excepting exceptions here but not raising it again, instead return null data from here.